<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="RVWidget" name="widgetRV">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>400</width>
          <height>300</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Utiliser la souris pour bouger le cube</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QPushButton" name="animationButton">
          <property name="toolTip">
           <string notr="true">Activer l'animation</string>
          </property>
          <property name="text">
           <string>Animation</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Réglages caméra</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Ouverture focale</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDial" name="fov">
             <property name="toolTip">
              <string>Change la focale de la caméra</string>
             </property>
             <property name="minimum">
              <number>2</number>
             </property>
             <property name="maximum">
              <number>120</number>
             </property>
             <property name="pageStep">
              <number>5</number>
             </property>
             <property name="value">
              <number>45</number>
             </property>
             <property name="tracking">
              <bool>true</bool>
             </property>
             <property name="notchTarget">
              <double>5.000000000000000</double>
             </property>
             <property name="notchesVisible">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Nouveau cube</string>
          </property>
          <widget class="QLabel" name="label_2">
           <property name="geometry">
            <rect>
             <x>60</x>
             <y>50</y>
             <width>67</width>
             <height>17</height>
            </rect>
           </property>
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="text">
            <string>y</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_3">
           <property name="geometry">
            <rect>
             <x>60</x>
             <y>20</y>
             <width>67</width>
             <height>17</height>
            </rect>
           </property>
           <property name="text">
            <string>x</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_4">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>110</y>
             <width>67</width>
             <height>17</height>
            </rect>
           </property>
           <property name="text">
            <string>roulis</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_5">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>140</y>
             <width>67</width>
             <height>17</height>
            </rect>
           </property>
           <property name="text">
            <string>tangage</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_6">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>170</y>
             <width>67</width>
             <height>17</height>
            </rect>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>lacet</string>
           </property>
          </widget>
          <widget class="QDoubleSpinBox" name="x_value">
           <property name="geometry">
            <rect>
             <x>80</x>
             <y>20</y>
             <width>63</width>
             <height>26</height>
            </rect>
           </property>
          </widget>
          <widget class="QDoubleSpinBox" name="y_value">
           <property name="geometry">
            <rect>
             <x>80</x>
             <y>50</y>
             <width>63</width>
             <height>26</height>
            </rect>
           </property>
          </widget>
          <widget class="QDoubleSpinBox" name="z_value">
           <property name="geometry">
            <rect>
             <x>80</x>
             <y>80</y>
             <width>63</width>
             <height>26</height>
            </rect>
           </property>
          </widget>
          <widget class="QDoubleSpinBox" name="roulis">
           <property name="geometry">
            <rect>
             <x>80</x>
             <y>110</y>
             <width>63</width>
             <height>26</height>
            </rect>
           </property>
          </widget>
          <widget class="QDoubleSpinBox" name="tangage">
           <property name="geometry">
            <rect>
             <x>80</x>
             <y>140</y>
             <width>63</width>
             <height>26</height>
            </rect>
           </property>
          </widget>
          <widget class="QDoubleSpinBox" name="lacet">
           <property name="geometry">
            <rect>
             <x>80</x>
             <y>170</y>
             <width>63</width>
             <height>26</height>
            </rect>
           </property>
          </widget>
          <widget class="QLabel" name="label_7">
           <property name="geometry">
            <rect>
             <x>60</x>
             <y>80</y>
             <width>67</width>
             <height>17</height>
            </rect>
           </property>
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="text">
            <string>z</string>
           </property>
          </widget>
          <widget class="QPushButton" name="pushButton">
           <property name="geometry">
            <rect>
             <x>30</x>
             <y>210</y>
             <width>89</width>
             <height>25</height>
            </rect>
           </property>
           <property name="text">
            <string>create cube</string>
           </property>
          </widget>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Minimum</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>RVWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>rvwidget.h</header>
   <slots>
    <slot>startAnimation()</slot>
    <slot>changeFov(int)</slot>
    <slot>changeOpacity(int)</slot>
    <slot>changeWireFrame(bool)</slot>
    <slot>changeCulling(bool)</slot>
    <slot>changeScale(int)</slot>
    <slot>changeSaturation(int)</slot>
    <slot>changeCameraType(bool)</slot>
    <slot>createCube()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>animationButton</sender>
   <signal>clicked()</signal>
   <receiver>widgetRV</receiver>
   <slot>startAnimation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>708</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>412</x>
     <y>423</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fov</sender>
   <signal>valueChanged(int)</signal>
   <receiver>widgetRV</receiver>
   <slot>changeFov(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>742</x>
     <y>157</y>
    </hint>
    <hint type="destinationlabel">
     <x>412</x>
     <y>522</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>createCube()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>711</x>
     <y>481</y>
    </hint>
    <hint type="destinationlabel">
     <x>720</x>
     <y>715</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>createCube()</slot>
 </slots>
</ui>
